doc/nrf/ug_multi_image.rst

```diff
@@ -3,160 +3,120 @@
+The parent image then *chain-loads* (or *boots*) the child image, which in turn might boot another child image.
+The most common use case for builds consisting of multiple images is an application that requires a bootloader to be present.
 
-When building the child image from source or using a prebuilt HEX
-file, the build system will merge the HEX files of the parent and
-child image together so that they can easily be programmed using
-the ``ninja flash`` target. This means that in the simplest scenario,
-it is possible to enable and integrate an additional image without
-even realizing it, using the same workflow as for additional
-libraries, namely enabling through Kconfig.
 
-Child images are configured with the same mechanisms as parent
-images: through CMake variables, Kconfig input fragments, and by
-modifying the ``.config`` file in the build directory. For example, to
-change the CONF_FILE for the MCUBoot image and the parent image, you
-could run the following command::
+When to use multiple images
+***************************
 
-   cmake -Dmcuboot_CONF_FILE=prj_a.conf -DCONF_FILE=app_prj.conf
+An *image* (also referred to as executable, program, or elf file) consists of pieces of code and data that are identified by image-unique names recorded in a single symbol table.
+The symbol table exists as metadata in a ``.elf`` or ``.exe`` file and is not included when the image is converted to a HEX file for programming.
+Instead, the code and data are placed at addresses by a linker.
+This linking process is what distinguishes images from object files (which do not require linking).
+Therefore, to determine if you have zero, one, or more images, count the number of times the linker runs.
 
-If you check the CONF_FILE, you can see that all CMake Cache
-variables that are image-specific are given a prefix to disambiguate
-them. To simplify configuation of single-image builds, the top-level
-image has the empty string as its prefix.
+Using multiple images has the following advantages:
 
-The same prefix convention is used to disambiguate targets. This means
-that to run menuconfig, for example, you can simply invoke the
-``menuconfig`` target to configure the top-level image and
-``mcuboot_menuconfig`` to configure the MCUBoot child image.
+* You can run the linker multiple times and partition the final firmware into several regions.
+  This partitioning is often useful for bootloaders.
+* Since there is a symbol table for each image, the same symbol names can exist multiple times in the final firmware.
+  This is useful for bootloader images, because it makes it possible to reuse libraries that the application image uses in the bootloader image.
+  (? is this correct? Original input was "This again is useful for bootloader images, as they often need to have their own fixed copies of the same libraries that the application image is using." ?)
```

Comment:
I wouldn't word it like this. reuse might imply the library only exists once in flash, but the whole point is that you might need the same library (same sources and symbol names at least) configured differently in two different places in flash.

Reply:
Might even be the "same" library, but two different versions, so not the exact same sources per se, but still the same problem of symbol name aliasing issues which require that the libraries be placed in two different images.

Reply:
Updated to: 
This is useful for bootloader images, which might required their own copy of a library that the application uses, but in a different version or configuration.